src/event-mixin.js

```diff
@@ -226,7 +258,28 @@ export function addEventListener(type, fn, optionsOrCapture) {
     this.__eventListenerCount = 0;
   }
   this.__eventListenerCount++;
-  let wrappedFn = function(e) {
+
+  // The function might be used for multiple events, so we need to keep track of
+  // the event type, options, and node into this object where the key determines
+  // the Id of the wrapped function (type+options), and the value is a WeakMap where
+  // we associate a node to the wrapped function.
+  fn.__eventWrappers = fn.__eventWrappers || {};
+  // This event listener might be added multiple times, we need to be able to remove
+  // all the wrappers we add.
+  const eventWrapperId = getEventWrapperId(type, optionsOrCapture);
+  const wrappersForType = fn.__eventWrappers[eventWrapperId] || new WeakMap();
```

Comment:
@sorvell I've tried few options & used this benchmark to measure how slow this patch is: synchronously add 1000 different event listeners, measure how long it takes https://gist.github.com/valdrinkoshi/32d33fd66b7cbcd48a9f70f30fb572ce
The native addEventListener takes ~2-3ms, the current approach takes ~8-10ms (2.5-3x more)
I've tried not relying on the WeakMap by generating the eventWrapperId like this:
```js
let nodeIdCount = 0;
function getEventWrapperId(node, type, options) {
  node.__id = node.__id || node.id || ++nodeIdCount;
  return node.__id + '|' + type + '|' + normalizedOptions(options);
}
```
This allows to have only 1 big object `fn.__eventWrappers` that we fill with these keys, and have `fn._eventWrappers[wrapperId] = wrappedFn`.
Performance got worse (~12-14ms, 5-6x slower).

WDYT?